src/atomEffect.ts

```diff
@@ -52,6 +52,10 @@ export function atomEffect(
         try {
           ref.cleanup?.()
           ref.cleanup = effectFn(get, setSelf as Setter)
+          if (!ref.mounted) {
+            ref.cleanup?.()
+            ref.cleanup = undefined
+          }
```

Comment:
Should we run effect if not mounted? How about this?
```js
ref.cleanup = ref.mounted ? effectFn(get, setSelf as Setter) : undefined
```

Reply:
this make sense

Reply:
if its not mounted, then I don't think the cleanup should fire either.

```ts
return (ref.promise = Promise.resolve().then(() => {
  try {
    if (!ref.mounted) return
    ref.cleanup?.()
    ref.cleanup = effectFn(get, setSelf as Setter)
  } finally {
    --ref.inProgress
    ref.promise = undefined
  }
}))
```